source/common/common/token_bucket_impl.cc

```diff
@@ -103,4 +103,16 @@ double AtomicTokenBucketImpl::timeNowInSeconds() const {
   return std::chrono::duration<double>(time_source_.monotonicTime().time_since_epoch()).count();
 }
 
+std::chrono::milliseconds AtomicTokenBucketImpl::nextTokenAvailable() const {
+  // If there are tokens available, return immediately.
+  if (remainingTokens() >= 1) {
+    return std::chrono::milliseconds(0);
+  }
+
+  // Calculate time since the last fill.
+  double current_time = timeNowInSeconds();
+  return std::chrono::milliseconds(
+      static_cast<uint64_t>(1 / fill_rate_ * 1000 - (current_time - time_in_seconds_.load())));
```

Comment:
I think the fill_rate_ is the number of tokens that will return to the bucket on each second. Then `1 / fill_rate_` would be required seconds to generate one token. Then we minus `(current_time - time_in_seconds_.load())` to get the seconds before next token?

If you want to convert the final output to ms, you should multiply 1000 with the `(1 / fill_rate_ - (current_time - time_in_seconds_.load())`?

Reply:
```
double AtomicTokenBucketImpl::timeNowInSeconds() const {
  return std::chrono::duration<double>(time_source_.monotonicTime().time_since_epoch()).count();
}
```
Am I missing something here?

Reply:
My bad, you're right.